tests/airgapped/README.md

```diff
@@ -0,0 +1,91 @@
+# Testing Airgapped Installation
+
+For running the tests we expect an environment that can:
+1. Spin up LXC containers
+2. Have Docker, to pull images
+
+We need docker to pull all the images necessary and push them to the airgapped
+lxc container.
+
+## Setup
+
+We've prepared some scripts for setting up the environment
+```bash
+./tests/airgapped/setup/setup.sh
+```
+
+We've observed that in almost all cases we needed to reboot to be able to
+run docker as sudo and the lxc container to, initially, have network access.
+
+## Running the tests
+
+You can run the script that will spin up an airgapped microk8s cluster with:
+
+```bash
+./tests/airgapped/airgap.sh \
+  --node-name airgapped-microk8s \
+  --microk8s-channel 1.24/stable
+  --bundle-path releases/latest/edge/bundle.yaml
+```
+
+### Running with a subset of images
+
+By default, if no `images.tar.gz` file is found, the script will try to download
+all the CKF images. These are 125Gb, which will make it difficult for running a
+lot of tests locally.
+
+Devs are urged to instead define their own `images.txt` file with the images
+they'd like to be loaded during tests. Note that in the instructions below I
+used `1.7/stable` until https://github.com/canonical/bundle-kubeflow/issues/679
+is resolved, and we'll be able to use other bundle files.
```

Comment:
We should make this easier for ourselves and anyone else.  Probably most people running these don't need to get every `seldonio/*` image for every server type.  I wonder if we can differentiate in our repos between required backbone images (`gcr.io/ml-pipeline/frontend`) and optional images like the seldon servers, the notebook servers, the katib images, etc

Reply:
I think the problem here is not removing some of the images but rather dynamically using more images, during testing.

For example to run our UATs we might need to use dedicated images that can test KF in an airgapped environment. For example right now a step of a test KFP pipeline downloads data from the internet. In the future to make those tests work in airgap we'll need to have a custom container image for that step, with data baked it, to avoid going to the internet.

Then regarding the initial comment on differentiating images, I think the time to implement this might not be worth it considering how frequently this is used. The only time I needed to update this list is when I wanted to
1. Test specific charms (i.e. KFP)
2. Create an `images.tar.gz` with a subset of images/charms